downloads/swan-lake-release-notes/swan-lake-2201.0.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,381 @@
+
+  ```ballerina
+  function fn(int y, int z) returns int {
+     function (int, int, int) returns int f = (x, y, z) => x + y + z; // error: redeclared symbols 'y' and 'z'
+   
+     int x = 34; // This is not in an overlapping scope, so, it does not result in an error.
+     return f(12, 32, 33);
+  }
+   
+  function createService(int age) returns
+         service object { public function getAgeInFiveYears(int age) returns int; } {
+     return service object {
+             public function getAgeInFiveYears(int age) returns int { // error: redeclared symbol 'age'
+                 return 5 + age;
+             }
+         };
+  }
+  ```
+
+- Fixed a bug that caused classes with all `final` fields of immutable types to be considered a `readonly class` (i,e., a subtype of `readonly`).
+
+Such a class can no longer be used in a context that expects a subtype of `readonly`.
+
+  ```ballerina
+  class Person {
+     final string name;
+     final string[] & readonly address;
+   
+     function init(string name, string[] address) {
+         self.name = name;
+         self.address = address.cloneReadOnly();
+     }
+   
+     function getAddress() returns string => string:'join(", ", ...self.address);
+  }
+   
+  public function main() {
+     Person person = new ("May", ["Palm Grove", "Colombo 3"]);
+    
+     // This, which was allowed previously results in an error now.
+     readonly readOnlyValue = person;
+  }
+  ```
+
+- Fixed a bug that resulted in invalid table lookups due to not distinguishing between `int`, `float`, and `decimal` zero.
+
+  ```ballerina
+  import ballerina/io;
+   
+  type Employee record {|
+     readonly int|float|decimal age;
+     int code;
+  |};
+   
+  public function main() {
+     table<Employee> key(age) empTable = table [
+         {age: 0, code: 100} // int `0` is the key
+     ];
+   
+     Employee? v1 = empTable[0f]; // Lookup with float `0` as the key.
+     io:println(v1 is Employee); // Previously `true`, now `false`.
+   
+     Employee? v2 = empTable[0d]; // Lookup with decimal `0` as the key.
+     io:println(v2 is Employee); // Previously `true`, now `false`.
+   
+     Employee? v3 = empTable[0]; // Lookup with int `0` as the key.
+     io:println(v3 is Employee); // Only this will continue to be `true`.
+  }
+  ```
+
+To view bug fixes, see the [GitHub milestone for Swan Lake 2201.0.0](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+Swan+Lake+-+2201.0.0%22+label%3AType%2FBug+label%3ATeam%2FCompilerFE).
+
+### Runtime Updates
+
+#### New Features
+
+#### Improvements
+
+#### Bug Fixes
+
+To view bug fixes, see the [GitHub milestone for Swan Lake 2201.0.0](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+Swan+Lake+-+2201.0.0%22+label%3AType%2FBug+label%3ATeam%2FjBallerina).
```

Comment:
```suggestion
To view bug fixes, see the [GitHub milestone for Swan Lake 2201.0.0](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AType%2FBug+label%3ATeam%2FjBallerina+milestone%3A%22Ballerina+2201.0.0%22).
```

Reply:
Made a slight edit to the above suggestion: Swan Lake 2201.0.0 -> 2201.0.0 (Swan Lake)